### PR TITLE
Creating a pull request with some guidance and examples.

### PR DESCRIPTION
**Pull request by gofforth**
Alrighty, made some changes and I'm going to annotate everything I did to try to explain. I tried not to do too much so that you can still work on it. 

We can talk about loading and saving to a file in another Pull Request.

### CODE REVIEW

File: Character.cs [hunk 1]
```diff
@@ -8,16 +8,26 @@ namespace ConsoleInitApp
 {
     class Character
     {
-        public string CharacterName { get; set; }
-        //public Action ActionTaken { get; set; }
-        public int HPValue { get; set; }
-        public int DamageValue { get; set; }
+        public string Name { get; set; }
```
Comment (gofforth) [inline Character.cs:11]: This was just a pet-peeve of mine. I like for property names to be as concise as possible. Additional words like Value don't add any useful information. And for `CharacterName` we are in the `Character` class its kind of implied.

File: Character.cs [hunk 2]
```diff
@@ -8,16 +8,26 @@ namespace ConsoleInitApp
...
+        public int HP { get; set; }
+        public int Damage { get; set; }
 
-        public Character(string characterName, int hpValue, int damageValue)
+        public Character()
```
Comment (gofforth) [inline Character.cs:15]: So here I just created what is often called a NULL constructor or a dummy constructor. This allows us to create an instance of an object without knowing the details ahead of time.

File: Character.cs [hunk 3]
```diff
@@ -8,16 +8,26 @@ namespace ConsoleInitApp
...
+
+        public Character(string name, int hp, int damage)
+        {
+            Name = name;
+            HP = hp;
+            Damage = damage;
+        }
+
+
+        public override string ToString()
```
Comment (gofforth) [inline Character.cs:28]: Created a `ToString()` method so we can print out the character.

File: Menu.cs [hunk 4]
```diff
@@ -22,7 +22,7 @@ public string Answer()
 
         public void Print()
         {
-            Console.WriteLine(Title);
+            Console.WriteLine("\n{0}\n-----------------------------",Title);
```
Comment (gofforth) [inline Menu.cs:25]: This was just a cosmetic thing to make the menu look more like a menu the format turns into:
```

<TITLE>
-----------------------------
```

File: Menu.cs [hunk 5]
```diff
@@ -31,6 +31,7 @@ public void Print()
 
         public int ReadOption()
         {
+            Console.Write("\n>: ");
```
Comment (gofforth) [inline Menu.cs:34]: Another cosmetic thing to show that the program is waiting for input.

File: Party.cs [hunk 6]
```diff
@@ -1,8 +1,5 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
```
Comment (gofforth) [inline Party.cs:3]: These we just unused default includes. So I got rid of them.

File: Program.cs [hunk 7]
```diff
@@ -1,18 +1,40 @@
...
 
 namespace ConsoleInitApp
 {
     class Program
     {
-       static bool shouldQuit = false;
-        static bool goBack = false;
+        static List<Character> Characters = new List<Character>();
+        static List<Party> Parties = new List<Party>();
+        static List<Encounter> Encounters = new List<Encounter>();
```
Comment (gofforth) [inline Program.cs:11]: We need a place to store the content we are generating in the menus.

File: Program.cs [hunk 8]
```diff
@@ -1,18 +1,40 @@
...
 
-        static void Main(string[] args)
+        static void Pause()
+        {
+            Console.Write("\nPress any key to continue...");
+            Console.ReadKey();
+        }
```
Comment (gofforth) [inline Program.cs:17]: Simple little convenience method. I think we had a version of it before.

File: Program.cs [hunk 9]
```diff
@@ -1,18 +1,40 @@
...
+
+        static string Prompt(string message)
         {
-            //Main menu to begin app. Simple choose option.
+            Console.WriteLine("\n{0}:", message);
+            Console.Write(">: ");
+            return Console.ReadLine();
+        }
```
Comment (gofforth) [inline Program.cs:24]: Method to prompt for information, it shows the prompt then returns the input.

File: Program.cs [hunk 10]
```diff
@@ -1,18 +1,40 @@
...
+        {
+            bool shouldQuit = false;
+            Menu start = new Menu("Main menu", new string[] {
+                "Add a character",
+                "Show characters",
+                "Add a party",
+                "Add an encounter",
+                "Start an Encounter",
+                "Quit"
+            });
```
Comment (gofforth) [inline Program.cs:37]: This was just a visual formatting thing so I could keep track of what options go where.

File: Program.cs [hunk 11]
```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
...
+                        AddParty();
                         break;
                     case 4:
-                        StartEncounter();
+                        AddEncounter();
                         break;
                     case 5:
+                        StartEncounter();
+                        break;
+                    case 6:
```
Comment (gofforth) [inline Program.cs:61]: These changes are due to me inserting a menu option in the middle.

File: Program.cs [hunk 12]
```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
...
                 }
             }
-          
         }
 
-
         static void AddCharacter()
         {
-            Menu CharacterMenu = new Menu("Choose an Option", new string[] { "Enter Name", "Enter HP", "Enter Damage" });
+            bool goBack = false;
```
Comment (gofforth) [inline Program.cs:72]: I moved the loop control variable inside the functions where they are used. This helps prevent them from being contaminated accidentally by other functions.

File: Program.cs [hunk 13]
```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
...
+            Menu CharacterMenu = new Menu("Add a character", new string[] {
+                "Enter Name",
+                "Enter HP",
+                "Enter Damage",
+                "Show Character",
+                "Go Back"
+            });
```
Comment (gofforth) [inline Program.cs:79]: More visual formatting.

File: Program.cs [hunk 14]
```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
...
 
-            while (!goBack) 
-            {
+            Character newCharacter = new Character();
+            Characters.Add(newCharacter);
```
Comment (gofforth) [inline Program.cs:82]: Here we create a new Character using that NULL constructor, and add it to our list of characters.

File: Program.cs [hunk 15]
```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
...
+            while (!goBack)
+            {
                 CharacterMenu.Print();
                 int option = CharacterMenu.ReadOption();
 
                 switch (option)
                 {
                     case 1:
-                        EnterName();
+                        EnterName(newCharacter);
```
Comment (gofforth) [inline Program.cs:92]: We now pass the newly created character to the functions that will update the various properties.

File: Program.cs [hunk 16]
```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
...
                     case 2:
-                        EnterXP();
+                        EnterXP(newCharacter);
                         break;
                     case 3:
-                        EnterDamage();
+                        EnterDamage(newCharacter);
                         break;
                     case 4:
+                        Console.WriteLine("\n{0}", newCharacter.ToString());
```
Comment (gofforth) [inline Program.cs:101]: Added an option to show the character we are currently adding.

File: Program.cs [hunk 17]
```diff
@@ -75,13 +111,20 @@ static void AddCharacter()
             }
         }
 
-        static void EnterName()
+        static void EnterName(Character character)
+        {
+            character.Name = Prompt("Enter the character's name");
+        }
```
Comment (gofforth) [inline Program.cs:117]: Here we are using our `Prompt()` helper to set the value of the the character's name property.

File: Program.cs [hunk 18]
```diff
@@ -75,13 +111,20 @@ static void AddCharacter()
...
+
+        static void EnterXP(Character character)
         {
-            Menu.Answer
-            
-            
+            character.HP = int.Parse(Prompt("Enter the character's HP"));
         }
```
Comment (gofforth) [inline Program.cs:122]: Ditto for HP.

File: Program.cs [hunk 19]
```diff
@@ -75,13 +111,20 @@ static void AddCharacter()
...
 
+        static void EnterDamage(Character character)
+        {
+            character.Damage = int.Parse(Prompt("Enter the character's damage"));
+        }
```
Comment (gofforth) [inline Program.cs:127]: Ditto for Damage.

File: Program.cs [hunk 20]
```diff
@@ -90,12 +133,24 @@ static void AddParty()
...
+
+        static void ShowCharacters()
+        {
+            Console.WriteLine("\nCharacters\n------------------------------");
+            foreach (Character c in Characters)
+            {
+                Console.WriteLine(c.ToString());
+            }
+            Pause();
+        }
```
Comment (gofforth) [inline Program.cs:152]: And this lets us print out all of the characters we've added.

**Review by Heinsolo — APPROVED** [review]

**Pull request closed without merging**